src/NHibernate.Test/Ado/BatcherFixture.cs

```diff
@@ -44,11 +43,11 @@ public void OneRoundTripInserts()
 		private void Cleanup()
 		{
 			using (ISession s = Sfi.OpenSession())
-			using (s.BeginTransaction())
+			using (var t = s.BeginTransaction())
 			{
 				s.CreateQuery("delete from VerySimple").ExecuteUpdate();
 				s.CreateQuery("delete from AlmostSimple").ExecuteUpdate();
-				s.Transaction.Commit();
+				t.Commit();
```

Comment:
I think it's better to isolate such changes in tests in separate commit to simplify review (at least in PR).
And you should add some Obsolete test that still uses `s.Transaction.Commit();` to make sure this case still works.

Reply:
That would mean a commit that does not even compile, since we treat warnings as errors. It is troublesome.

Reply:
Just mark test as `[Obsolete]` and it should be OK

Reply:
What? These tests are not obsolete, they need to be updated. We have to keep them, not drop them at next major release.

Reply:
You:
> That would mean a commit that does not even compile, since we treat warnings as errors. It is troublesome.

Me:
> Just mark test as [Obsolete] and it should be OK

And you've just committed tests marked as `[Obsolete]` for legacy `s.Transaction`... That's all I meant.. So not sure what's you `What?` about..

Reply:
Misunderstanding all the way long:
The following:

> That would mean a commit that does not even compile, since we treat warnings as errors. It is troublesome.

Was refeting to:

> I think it's better to isolate such changes in tests in separate commit to simplify review (at least in PR).

So when you replied:

> Just mark new test as `[Obsolete]` and it should be OK

I thought you were willing me to flag as obsolete all the changed tests to avoid altering them an "ease" the review. Thus my reaction.

So the trouble remain: how do you want to split in separated commits the fix of the trouble and the fix of the tests, without issuing a commit that cannot compile?

Reply:
I don't understand why non-compilable commit is an issue here. When you push multiple commits at once - only last one will be compiled.

Reply:
Just a principle I think should be followedto (and I do not think I am alone): do not commit something that does not compile.
A separated commit should always compile. It can be cherry-picked, we can branch from it, ... It is not really "nice" to leave a commit which does not compile, even in a tree just meant for PR review.

Reply:
Done